src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -743,17 +736,16 @@ static bool parseObject (
         case STI_ARRAY:
             try
             {
-                std::unique_ptr <STArray> sub_array_;
-                bool const success (parseArray (json_name + "." + fieldName,
-                    value, field, depth + 1, sub_array_, error));
-                if (! success)
+                auto array = parseArray (json_name + "." + fieldName,
+                    value, field, depth + 1, error);
+                if (array == boost::none)
                     return false;
```

Comment:
Shouldn't this be `return boost::none`?  Why does this compile?  Since it compiles, what is it returning?